MuxExoPlayer/src/r2_14_1/java/com/mux/stats/sdk/muxstats/MuxStatsExoPlayer.java

```diff
@@ -0,0 +1,436 @@
+              loadEventInfo.uri.getPath(), mediaLoadData.dataType,
+              loadEventInfo.uri.getHost(), segmentMimeType);
+    } else {
+      MuxLogger.d(TAG,
+          "ERROR: onLoadStarted called but mediaLoadData argument have no uri parameter.");
+    }
+  }
+
+  @Override
+  public void onMetadata(AnalyticsListener.EventTime eventTime, Metadata metadata) {
+  }
+
+  @Override
+  public void onPlaybackParametersChanged(AnalyticsListener.EventTime eventTime,
+      PlaybackParameters playbackParameters) {
+    onPlaybackParametersChanged(playbackParameters);
+  }
+
+  @Override
+  public void onPlaybackStateChanged(EventTime eventTime, @Player.State int state) {
+    onPlaybackStateChanged(state);
+  }
+
+  @Override
+  public void onPlaybackSuppressionReasonChanged(AnalyticsListener.EventTime eventTime,
+      int playbackSuppressionReason) {
+  }
+
+  @Override
+  public void onPlayerError(AnalyticsListener.EventTime eventTime, ExoPlaybackException error) {
+    onPlayerError(error);
+  }
+
+  @Override
+  public void onPlayWhenReadyChanged(AnalyticsListener.EventTime eventTime, boolean playWhenReady,
+      int reason) {
+    onPlayWhenReadyChanged(playWhenReady, reason);
+    onPlaybackStateChanged(player.get().getPlaybackState());
+  }
+
+  @Override
+  public void onPositionDiscontinuity(AnalyticsListener.EventTime eventTime, int reason) {
+    onPositionDiscontinuity(reason);
+  }
+
+  @Override
+  public void onRepeatModeChanged(AnalyticsListener.EventTime eventTime, int repeatMode) {
+    onRepeatModeChanged(repeatMode);
+  }
+
+  // Note: onSeekProcessed was deprecated in 2.12.0
+
+  @Override
+  public void onSeekStarted(AnalyticsListener.EventTime eventTime) {
+    seeking();
+  }
+
+  @Override
+  public void onShuffleModeChanged(AnalyticsListener.EventTime eventTime,
+      boolean shuffleModeEnabled) {
+    onShuffleModeEnabledChanged(shuffleModeEnabled);
+  }
+
+  @Override
+  public void onSurfaceSizeChanged(AnalyticsListener.EventTime eventTime, int width,
+      int height) {
+  }
+
+  @Override
+  public void onTimelineChanged(AnalyticsListener.EventTime eventTime, int reason) {
+    onTimelineChanged(eventTime.timeline, reason);
+  }
+
+  @Override
+  public void onTracksChanged(AnalyticsListener.EventTime eventTime, TrackGroupArray trackGroups,
+      TrackSelectionArray trackSelections) {
+    onTracksChanged(trackGroups, trackSelections);
+  }
+
+  @Override
+  public void onUpstreamDiscarded(EventTime eventTime, MediaLoadData mediaLoadData) {
+  }
+
+  @Override
+  public void onVideoSizeChanged(AnalyticsListener.EventTime eventTime, int width, int height,
+      int unappliedRotationDegrees, float pixelWidthHeightRatio) {
+    sourceWidth = width;
+    sourceHeight = height;
+  }
+
+  @Override
+  public void onVolumeChanged(AnalyticsListener.EventTime eventTime, float volume) {
+  }
+  // ------END AnalyticsListener callbacks------
+
+  // ------BEGIN Player.EventListener callbacks------
```

Comment:
In line with my previous question, we seem to have reclassified at compile time to use ExoPlayer.EventListener (for a reason I'm not certain about), so I'm interested if this causes any compatibility issues on older versions.